cmd/istioctl/main.go

```diff
@@ -316,3 +323,39 @@ func readInputs() ([]inputDoc, error) {
 
 	return varr, nil
 }
+
+// Print a simple list of names
+func printShortOutput(list map[model.Key]proto.Message) error {
+	for key := range list {
+		fmt.Printf("%v\n", key.Name)
+	}
+
+	return nil
+}
+
+// Print as YAML
+func printYamlOutput(list map[model.Key]proto.Message) error {
+	var retVal error
+
+	for key, item := range list {
+		out, err := schema.ToYAML(item)
+		if err != nil {
+			retVal = err // Save the error, but keep going, so we can print the valid rules that follow
```

Comment:
This will only report the most recent error. Perhaps you could use https://github.com/hashicorp/go-multierror if you want to collect all of the errors?

Also, I wouldn't print the error value here, but instead let the caller handle it accordingly, e.g. after normal output is printed and not interleaved with successful output.